radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/RadixClientSupplier.java

```diff
@@ -0,0 +1,177 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Given a network, a selector and filters, yields the {@link RadixJsonRpcClient} to use
+ */
+public class RadixClientSupplier {
+	private static final Logger LOGGER = LoggerFactory.getLogger(RadixClientSupplier.class);
+
+	/**
+	 * The amount of time to delay in between node connection requests
+	 */
+	private final int delaySecs;
+
+	/**
+	 * The network of peers available to connect to
+	 */
+	private final RadixNetwork network;
+
+	/**
+	 * The selector to use to decide between a list of viable peers
+	 */
+	private final RadixPeerSelector selector;
+
+	/**
+	 * The filters to test whether a peer is desirable
+	 */
+	private final List<RadixPeerFilter> filters;
+
+	/**
+	 * Create a client selector from a certain network with the default filters and the a randomized selector
+	 * @param network The network
+	 */
+	public RadixClientSupplier(RadixNetwork network) {
+		this(network, new RandomSelector(), Arrays.asList(
+				new ConnectionAliveFilter(),
+				new CompatibleApiVersionFilter(RadixJsonRpcClient.API_VERSION)));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a matching universe filter, the default filters and a randomized selector
+	 * @param network The network
+	 * @param config The universe config for the matching universe filter
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixUniverseConfig config) {
+		this(network, new RandomSelector(), Arrays.asList(
+				new ConnectionAliveFilter(),
+				new CompatibleApiVersionFilter(RadixJsonRpcClient.API_VERSION),
+				new MatchingUniverseFilter(config)));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a certain selector and filters
+	 * @param network The network
+	 * @param selector The selector used to select a peer from the desirable peer list
+	 * @param filters The filters used to test the desirability of peers
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixPeerSelector selector, RadixPeerFilter... filters) {
+		this(network, selector, Arrays.asList(filters));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a certain selector and filters
+	 * @param network The network
+	 * @param selector The selector used to select a peer from the desirable peer list
+	 * @param filters The filters used to test the desirability of peers
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixPeerSelector selector, List<RadixPeerFilter> filters) {
+		Objects.requireNonNull(network, "network is required");
+		Objects.requireNonNull(selector, "selector is required");
+		Objects.requireNonNull(filters, "filters is required");
+
+		this.network = network;
+		this.selector = selector;
+		this.filters = filters;
```

Comment:
`filters` here should probably be copied.

Reply:
Now is copied into immutable list